### PR TITLE
Adding pysciencemode

### PR DESCRIPTION
**Pull request by Kev1CO**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by Kev1CO** [comment]
@conda-forge/help-python

**Comment by Kev1CO** [comment]
Alike other pull requests, I have an issue with Build osx osx_64.
If you have any information on how to fix it, it would be appreciated.
Have a good day !

File: recipes/pyScienceMode/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  version: "{{ version }}"
+
+source:
+  url: https://files.pythonhosted.org/packages/51/a4/14e4d9e859a0109c4e44c36f7e759ae8f3d24a603c72ddf28e9ee9919566/pysciencemode-1.1.4.tar.gz
+  sha256: 91952b59df0ab6b7472c302fd0b5777e271958e57c48d00d063ca1769faa70e6
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
```
Comment (matthewfeickert) [inline recipes/pyScienceMode/meta.yaml:16]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```

Consider using `--no-deps --no-build-isolation` to ensure that all of the required build system dependencies are included as `host` dependencies. (I'm not a conda-forge/core maintainer, I'm just a drive by person who happened to be checking to make sure other PRs were failing on macOS. :+1:)

File: recipes/pyScienceMode/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{% set name = "pysciencemode" %}
+{% set version = "1.1.4" %}
+{% set python_min = "3.10" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://files.pythonhosted.org/packages/51/a4/14e4d9e859a0109c4e44c36f7e759ae8f3d24a603c72ddf28e9ee9919566/pysciencemode-1.1.4.tar.gz
```
Comment (matthewfeickert) [inline recipes/pyScienceMode/meta.yaml:10]:
```suggestion
  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/pysciencemode-{{ version }}.tar.gz
```

This is a strange URL: https://files.pythonhosted.org/ Why is this being used and not `pypi.org` given that https://pypi.org/project/pysciencemode/ exists? If you generate the recipe with

```
grayskull pypi pysciencemode
```

you'll get 

```yaml
  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/pysciencemode-{{ version }}.tar.gz
```

**Review by matthewfeickert — CHANGES_REQUESTED** [review]

**Comment by Kev1CO** [comment]
@matthewfeickert Thanks for the advices !
The osx_64 problem seems to hit several PR.

**Review by matthewfeickert — APPROVED** [review]

File: recipes/pyScienceMode/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  version: "{{ version }}"
+
+source:
+  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/pysciencemode-{{ version }}.tar.gz
+  sha256: 91952b59df0ab6b7472c302fd0b5777e271958e57c48d00d063ca1769faa70e6
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```
Comment (matthewfeickert) [inline recipes/pyScienceMode/meta.yaml:16]:
```suggestion
  script: {{ PYTHON }} -m pip install . -vv
```

I realize I gave you unnecessary advice earlier, as these get set automatically by the build tool. (c.f. https://github.com/conda/grayskull/issues/582)

Comment (Kev1CO) [reply recipes/pyScienceMode/meta.yaml]: Thanks for noticing it !

File: recipes/pyScienceMode/meta.yaml (hunk 1 above)
Comment (matthewfeickert) [reply recipes/pyScienceMode/meta.yaml:16]: @Kev1CO sorry it seems the recommendation is to add it back given https://github.com/conda-forge/staged-recipes/pull/28268#discussion_r1850668020. I'm going to stop commenting on this PR though given that I seem to have given you conflicting advice twice now.

File: recipes/pyScienceMode/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+build:
+  number: 0
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv
+
+
+requirements:
+  host:
+    - python >={{ python_min }}
```
Comment (xhochy) [inline recipes/pyScienceMode/meta.yaml:21]:
```suggestion
    - python {{ python_min }}
```

Comment (matthewfeickert) [reply recipes/pyScienceMode/meta.yaml]: Sorry I missed that earlier. :grimacing:

Comment (Kev1CO) [reply recipes/pyScienceMode/meta.yaml]: No worries 😃

**Review by xhochy — APPROVED** [review]

**Pull request merged**